plotly/tools.py

```diff
@@ -2509,6 +2561,205 @@ def create_dendrogram(X, orientation="bottom", labels=None,
         return {'layout': dendrogram.layout,
                 'data': dendrogram.data}
 
+    @staticmethod
+    def create_annotated_heatmap(z, x=None, y=None, annotation_text=None,
+                                 colorscale=None, fontcolor=None,
+                                 showscale=False, reversescale=False,
+                                 **kwargs):
+        """
+        BETA function that creates annotated heatmaps
+
+        This functions adds annotations to each cell of the heatmap.
+
+        :param (list[list]) z: z matrix to create heatmap.
+        :param (list) x: x axis labels.
+        :param (list) y: y axis labels.
+        :param (list[list]) text: Text strings for annotations. Should be the
+            same dimmensions as the z matrix. If no text is added, the the
+            values of the z matrix are annotated. Default = z matrix values.
+        :param (list) fontcolor: List of two colorstrings: [min_text_color,
```

Comment:
`color strings`? is `colorstring` something we define somewhere? (Sorry if I missed it).